tests/processing/test_harvesting.py

```diff
@@ -270,41 +275,85 @@ def test_scientific_quality_change(self, request_mock, fetch_mock):
         self.harvester().harvest_layers()
         self.assertEqual(DBSession.query(Layer).count(), 1)
+        self.assertEqual(DBSession.query(HazardSet).count(), 1)
 
         self.harvester().harvest_layers()
         self.assertEqual(DBSession.query(Layer).count(), 0)
+        self.assertEqual(DBSession.query(HazardSet).count(), 0)
 
     @patch.object(
         Harvester,
         "fetch",
         side_effect=[
             layers([{"id": 1}, {"id": 2}]),
-            layers()
+            layers([{"id": 1}])
         ]
     )
     @patch.object(
         httplib2.Http,
         "request",
         side_effect=[
-            (Mock(status=200), json.dumps(layer({"id": 1}))),
-            (Mock(status=200), json.dumps(layer({"id": 2})))
+            (Mock(status=200), json.dumps(layer({"id": 1, "hazard_period": 10}))),
+            (Mock(status=200), json.dumps(layer({"id": 2, "hazard_period": 50}))),
+            (Mock(status=200), json.dumps(layer({"id": 1, "hazard_period": 10}))),
         ]
     )
-    def test_layer_harvested_multi(self, request_mock, fetch_mock):
-        """Valid layers must be added to database and deleted if not harvested"""
+    def test_layer_removed(self, request_mock, fetch_mock):
+        """Layer removed from geonode should be removed from database"""
         self.harvester().harvest_layers()
-        self.assertEqual(DBSession.query(Layer).count(), 1)
+        self.assertEqual(DBSession.query(Layer).count(), 2)
+        self.assertEqual(DBSession.query(HazardSet).count(), 1)
 
+        hazardset = DBSession.query(HazardSet).one()
+        hazardset.completed = True
+        hazardset.processed = datetime.now()
+        DBSession.flush()
+
+        self.harvester().harvest_layers()
+        self.assertEqual(DBSession.query(Layer).count(), 1)
         self.assertEqual(DBSession.query(HazardSet).count(), 1)
+        self.assertFalse(hazardset.completed)
+        self.assertIsNone(hazardset.processed)
 
+    @patch.object(
+        Harvester,
+        "fetch",
+        side_effect=[
+            layers([{"id": 1}, {"id": 2}]),
+            layers([{"id": 1}, {"id": 2}])
+        ]
+    )
+    @patch.object(
+        httplib2.Http,
+        "request",
+        side_effect=[
+            (Mock(status=200), json.dumps(layer({"id": 1, "hazard_period": 15}))),
+            (Mock(status=200), json.dumps(layer({"id": 2, "hazard_period": 10}))),
+            (Mock(status=200), json.dumps(layer({"id": 1, "hazard_period": 15}))),
+            (Mock(status=200), json.dumps(layer({"id": 2, "hazard_period": 10}))),
+        ]
+    )
+    def test_no_useless_process_invalidate(self, request_mock, fetch_mock):
+        """Multiple layers matching same level should not invalidate processing uselessly"""
         self.harvester().harvest_layers()
-        self.assertEqual(DBSession.query(Layer).count(), 0)
+        self.assertEqual(DBSession.query(Layer).count(), 1)
```

Comment:
Do we have 1 layer here and not 2, because one supersedes the other?

Reply:
Yes, it is exactly what I want to test.

First harvesting:
- Layer1 match level HIG
- Layer2 match level HIG
=> we keep only Layer2

Second harvesting (nothing has changed on geonode side):
- Layer1 match level HIG
- Layer2 match level HIG
=> we should not have set hazadset.processed to None